#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/Parquetry.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.3035115374454458 -0.753856918297189 -0.582735354490327 5.068637274971969
  position -0.20830685813936603 -0.2295329785694687 0.4522058701392607
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 0.000353253
  floorSize 3 3
  floorAppearance Parquetry {
    type "light strip"
  }
}
Robot {
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        position -4.096000000000003
        axis 0 0 1
        anchor 0.00369351 0.02804 0.107
      }
      device [
        PositionSensor {
          name "yaw_sensor"
        }
        RotationalMotor {
          name "yaw_motor"
        }
      ]
      endPoint Solid {
        translation 0.00369351 0.02804 0.095
        rotation 0 0 -1 3.4415606464103305
        children [
          Solid {
            translation 0 0 0.027
            children [
              HingeJoint {
                jointParameters HingeJointParameters {
                  position 2.978560000000003
                  axis 0 1 0
                  anchor 0 0 0.02
                }
                device [
                  RotationalMotor {
                    name "pitch_motor"
                  }
                  PositionSensor {
                    name "pitch_sensor"
                  }
                ]
                endPoint Solid {
                  translation -0.014623674387905204 -0.015000000000000003 0.01666111584559672
                  rotation 0 1 0 2.917120000000005
                  children [
                    Shape {
                      appearance Appearance {
                        material Material {
                          diffuseColor 0.607843 0.607843 0.0705882
                        }
                      }
                      geometry Box {
                        size 0.05 0.01 0.01
                      }
                    }
                  ]
                  name "turret_vertical_arm"
                }
              }
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.243137 0.8 0.8
                  }
                }
                geometry Box {
                  size 0.02 0.02 0.05
                }
              }
            ]
            name "turret_main_arm"
          }
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 0.3 0.5
                shininess 0.5
              }
            }
            geometry Box {
              size 0.09 0.11 0.005
            }
          }
        ]
        name "horizontal_rotation_plate"
      }
    }
    Solid {
      translation 0 0 -0.01
      rotation 0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0354056994857648
      children [
        CadShape {
          url [
            "../assets/turret_tripod.obj"
          ]
        }
      ]
      name "base"
    }
  ]
  name "turret_bot"
  controller "turret_manual"
}
